alcs/acs2/ACS2.py

```diff
@@ -1,10 +1,16 @@
-from alcs.acs2 import ClassifiersList
+from alcs.acs2 import ClassifiersList, default_configuration, ACS2Configuration
 
-DO_GA = False
+DO_GA = True
 
 
 class ACS2:
-    def __init__(self, population=None):
+    def __init__(self,
+                 configuration: ACS2Configuration = default_configuration,
+                 population=None):
+        if configuration:
```

Comment:
You are using a default value in construction. Is `if-else` required?

Reply:
You are right, actually no in this case.
But actually I think it makes sense to do it a bit other way instead: to use default value == None, and in case of None initialize cfg with a _copy_ of default_configuration (so that any user code don't change the default values accidentally).

Reply:
Implemented it everywhere using following pattern:
`def __init__(self, cfg=None):
    self.cfg = cfg or ACS2Configuration.default()`

Also made sure that every object that constructs another object relying on configuration passes its own configuration.